src/Lecture1.hs

```diff
@@ -29,98 +29,38 @@ module Lecture1
     , subString
     , strSum
     , lowerAndGreater
-    ) where
+	, check
+    ) where 
 
--- VVV If you need to import libraries, do it after this line ... VVV
-
--- ^^^ and before this line. Otherwise the test suite might fail  ^^^
-
-{- | Specify the type signature of the following function. Think about
-its behaviour, possible types for the function arguments and write the
-type signature explicitly.
--}
+makeSnippet::Int -> String -> String
 makeSnippet limit text = take limit ("Description: " ++ text) ++ "..."
 
-{- | Implement a function that takes two numbers and finds sum of
-their squares.
-
->>> sumOfSquares 3 4
-25
-
->>> sumOfSquares (-2) 7
-53
-
-Explanation: @sumOfSquares 3 4@ should be equal to @9 + 16@ and this
-is 25.
--}
--- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-sumOfSquares x y = error "TODO!"
+sumOfSquares:: Int -> Int -> Int
+sumOfSquares x y = (x^2) + (y ^2)
 
-{- | Implement a function that returns the last digit of a given number.
+lastDigit::Int->Int
+lastDigit x = abs $ x `rem` 10
 
->>> lastDigit 42
-2
->>> lastDigit (-17)
-7
+minmax::Int -> Int -> Int-> Int
+minmax x y z = let (max,min) = (maximum [x,y,z],minimum [x,y,z]) in (max - min)
 
-🕯 HINT: use the @mod@ function
+subString:: Int -> Int-> String -> String
+subString start end str
+    | end < 0 || end < start = ""
+    | start < 0              = subString 0 end str
+    | otherwise              = take (end - start + 1) $drop start str
 
--}
--- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
-
-{- | Write a function that takes three numbers and returns the
-difference between the biggest number and the smallest one.
-
->>> minmax 7 1 4
-6
-
-Explanation: @minmax 7 1 4@ returns 6 because 7 is the biggest number
-and 1 is the smallest, and 7 - 1 = 6.
-
-Try to use local variables (either let-in or where) to implement this
-function.
--}
-minmax x y z = error "TODO"
+strSum :: String -> Int
+strSum "" = 0
+strSum str = sum(map read $ words str ::[Int])
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
strSum str = sum (map read (words str))
```